shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -38,8 +44,19 @@ type instance PreviousEra (AllegraEra c) = ShelleyEra c
 -- to provide the context in the right place.
 type instance TranslationContext (AllegraEra c) = ()
 
-instance Crypto c => TranslateEra (AllegraEra c) NewEpochState where
-  translateEra _ = error "TODO Shelley to Allegra translation"
+instance (ShelleyBased (AllegraEra c), Crypto c) => TranslateEra (AllegraEra c) NewEpochState where
```

Comment:
@mrBliss is it okay that I've add the `ShelleyBased` constraint? I suspect that we can remove it once `AllegraEra` is fleshed out. Or I can remove it now and just comment out the use of `returnRedeemAddrsToReserves`.

Reply:
Hmm, I would indeed expect that all of the constraints implied by `ShelleyBased` hold for `AllegraEra c`. This means that some constraints we expect to hold for each era are currently not implemented for `AllegraEra`.

I suppose it's fine to add it *for now* (maybe add a TODO comment so we don't forget about it). Hopefully GHC will warn us that the constraint is redundant as soon it is implemented. (In the end, this constraint should definitely go.)

P.S. Thanks for checking this with me, much appreciated!

Reply:
ok, sounds good!

Reply:
Yeah, we're still (until #1930) missing a `TxBody` for the ShelleyMA eras. Once that's in place we should have `ShelleyBased` for both Mary and Allegra